plugins/uefi/fu-uefi-bootmgr.c

```diff
@@ -321,11 +321,32 @@ fu_uefi_bootmgr_bootnext (const gchar *esp_path,
 	if (source_app == NULL)
 		return FALSE;
 
+	/* test if correct asset in place */
+	target_app = fu_uefi_get_esp_app_path (esp_path, "fwupd", error);
+	if (target_app == NULL)
+		return FALSE;
+	if (!fu_uefi_cmp_asset (source_app, target_app)) {
+		if (!fu_uefi_copy_asset (source_app, target_app, error))
+			return FALSE;
+	}
+
```

Comment:
Any particular reason this was moved up?  The main reason it was previously after the shim test was to not copy files if you know it's not going to work. (eg shim doesn't exist, secure boot is on, and flag is set to use shim for SB).

Reply:
It's so we can do `filepath = target_app;` -- let me rework things a little to avoid moving that block.

Reply:
Fixed.